gen_bootloader.sh

```diff
@@ -8,12 +8,36 @@ set_bootloader() {
 		grub2)
 			set_bootloader_grub2
 			;;
+		systemd-boot)
+			set_bootloader_systemd
+			;;
 		*)
 			print_warning "Bootloader ${BOOTLOADER} is not currently supported"
 			;;
 	esac
 }
 
+set_bootloader_systemd() {
+
+	if [[ ! -x `which kernel-install` ]]; then
+		gen_die "Unable to locate kernel-install, is systemd installed?"
+	fi
+
+	# Clear out the mistakenly installed copies, since kernel-install copies its own versions over
+	# We do this first, to prevent people thinking they have enough free space but needing twice the amount
```

Comment:
compile_kernel() in gen_compile.sh should be modified:
```diff
@@ -416,19 +416,24 @@ compile_kernel() {

        if isTrue "${CMD_INSTALL}"
        then
-               copy_image_with_preserve "kernel" \
-                       "${tmp_kernel_binary}" \
-                       "kernel-${KNAME}-${ARCH}-${KV}"
+               if [ -n "${BOOTLOADER}" -a "${BOOTLOADER}" = "systemd" ]
+               then
+                       print_info 1 "$(getIndent 1)Not installing kernel; systemd's kernel-install will handle installation for us!"
+               else
+                       copy_image_with_preserve "kernel" \
+                               "${tmp_kernel_binary}" \
+                               "kernel-${KNAME}-${ARCH}-${KV}"

-               copy_image_with_preserve "System.map" \
-                       "${systemmap}" \
-                       "System.map-${KNAME}-${ARCH}-${KV}"
+                       copy_image_with_preserve "System.map" \
+                               "${systemmap}" \
+                               "System.map-${KNAME}-${ARCH}-${KV}"

-               if isTrue "${GENZIMAGE}"
-               then
-                       copy_image_with_preserve "kernelz" \
-                               "${tmp_kernel_binary2}" \
-                               "kernelz-${KV}"
+                       if isTrue "${GENZIMAGE}"
+                       then
+                               copy_image_with_preserve "kernelz" \
+                                       "${tmp_kernel_binary2}" \
+                                       "kernelz-${KV}"
+                       fi
                fi
        else
                cp "${tmp_kernel_binary}" "${TMPDIR}/kernel-${KNAME}-${ARCH}-${KV}" ||
```

Reply:
This feels like it's moving bootloader logic into the compile_kernel logic.  I'm happy to do that, I just figured you'd want it all stand-alone in its own little area, so that the it doesn't set a precedent for the next bootloader that comes along to start tinkering with the internals too?

Reply:
You are right. The suggestion was based on genkernel-3, before I decided to touch so much code and create genkernel-4. Now I think we can do it correctly: But before we are able to do that we need to tackle https://bugs.gentoo.org/505810 - i.e. I'll remove install logic from `compile_*` to *genkernel* main script. In the main script we will keep something like https://github.com/gentoo/genkernel/blob/v4.0.0_beta4/genkernel#L337, adjusted of course, which will call 1set_bootloader1 when genkernel was called with `--install`. We will use the case statement currently used to display a warning to decide if we have to call `copy_image_with_preserve` or not... then, when we finally call `set_bootloader`, which would call `kernel-install` in systemd-boot case.

You get my idea?

tl;dr
Before we get back to adding systemd-boot support, I'll have to refactor install logic. :)

Reply:
Yeah, I think I follow.  Just give me a prod on here when it's ready for some reworking.  I'm happy to pitch in, but don't want to step on your plans...  5:)